src/main/java/org/mondemand/Client.java

```diff
@@ -550,7 +553,7 @@ public void increment (StatType type, String key, int value) {
     if(realValue == null) {
       // create the counter if doesn't exist
       realValue = new StatsMessage(realKey, type);
-      this.stats.put(realKey, realValue);
+      this.stats.putIfAbsent(realKey, realValue);
```

Comment:
any reason why not assigning realValue = this.stats.get(realKey) after the putIfAbsent?  the potential for loosing stats is very small, so there's probably little harm in this, but it's inconsistent with other similar bits in this class